lib/createWorkers.js

```diff
@@ -1,350 +1,19 @@
-    try {
-      const urlWithParam = url.resolve(amqpUrl, `?heartbeat=${heartbeat}`);
-      const connection = yield amqplib.connect(urlWithParam, socketOptions);
-      logger.info({ workerName, heartbeat }, '[AMQP] connected to server');
-      subscribeToConnectionEvents(connection, workerName);
-      return connection;
-    } catch (err) {
-      logger.error(
-        { err, heartbeat, socketOptions, workerName },
-        '[worker#_connectAmqp] connection failed'
-      );
-      throw err;
-    }
-  }
-
-  /**
-   * Create a channel for the worker
-   * @param {Object} connection the connection object
-   * @param {String} exchangeName the exchange name to use
-   * @param {String} queueName the queue name to target
-   * @param {String} routingKey routing key to bind on
-   * @param {Number} channelPrefetch number of message to prefetch from channel
-   * @param {String} workerName the name of the worker (for logs)
-   * @returns {Object} channel object
-   * @private
-   */
-  function* _createChannel(connection,
-    { exchangeName, queueName, channelPrefetch, workerName }) {
-    const channel = yield connection.createChannel();
-    logger.info(
-      { exchangeName, queueName, channelPrefetch, workerName },
-      '[AMQP] create channel'
-    );
-    subscribeToChannelEvents(
-      channel,
-      { exchangeName, queueName, channelPrefetch, workerName }
-    );
-    yield channel.prefetch(channelPrefetch);
-    yield channel.assertExchange(exchangeName, DEFAULT_EXCHANGE_TYPE);
-    yield channel.assertQueue(queueName, {});
-    return channel;
-  }
-
-/**
- * Bind a message handler on a queue to consume
- * @param   {Object} handler the handler content
- * @param   {function} handler.handle the message handler
- * @param   {function} handler.validate the message validator
- * @param   {String} handler.routingKey the routingKey to bind to
- * @returns {*} -
- * @private
- */
-  function* _bindHandler({ routingKey, handle, validate }) {
-    const { exchangeName, queueName, channelPrefetch, workerName } = workerConfig;
-    const formattedQueueName = _getQueueName(queueName, routingKey);
-
-    const workerChannel = yield _createChannel(workerConnection, {
-      exchangeName,
-      queueName: formattedQueueName,
-      channelPrefetch,
-      workerName
-    });
-    workerChannels.push(workerChannel);
-
-    yield workerChannel.bindQueue(formattedQueueName, exchangeName, routingKey);
-    yield workerChannel.consume(
-      formattedQueueName,
-      co.wrap(_getMessageConsumer(workerChannel, handle, validate))
-    );
-  }
-}
-
-/**
- * Returns the formatted queue name.
- *
- * It contains the base queue name and the routing key.
- *
- * @param       {String} queueName The base queue name.
- * @param       {String} routingKey The routing key.
- * @returns     {String} The formatted queue name.
- */
-function _getQueueName(queueName, routingKey) {
-  return `${queueName}.${routingKey}`;
+  options.logger = logger;
```

Comment:
you should not modify the `options` parameter, this side effect can be completely avoided 🙂

Reply:
ok